src/main/java/hillelee/doctor/DoctorService.java

```diff
@@ -52,4 +60,44 @@ public Boolean doctorExists(Integer id) {
         return doctorRepository.findAll().stream().anyMatch(doc -> doc.getId().equals(id));
     }
 
+    public Map<LocalTime, Integer> getDoctorsSchedule(Integer id, LocalDate date) {
+        if (!doctorRepository.findById(id).isPresent()) {
+            throw new NoSuchDoctorException();
+        }
+
+        Map<LocalTime, Integer> map = doctorRepository.findById(id)
+                .get()
+                .getAppointments().stream()
+                .filter(appointment -> appointment.getComingDate().equals(date))
+                .collect(Collectors.toMap(Appointment::getTime, Appointment::getPetId, (a, b) -> b));
+
+        return map.entrySet()
+                .stream()
+                .sorted(Map.Entry.comparingByKey())
+                .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue,
+                (oldValue, newValue) -> oldValue, LinkedHashMap::new));
+
+    }
+
+    public Doctor createAnAppointment(Integer id, LocalDate date, LocalTime time, Integer petId) {
```

Comment:
in real life it is highly probable that two pepole will create appointment to same doctor at same time